forch/forchestrator.py

```diff
@@ -418,7 +418,22 @@ def _register_handlers(self):
                 event.timestamp, event.dp_name, event.port_no, event.eth_src, event.l3_src_ip)),
             (FaucetEvent.L2Expire, lambda event: fcoll.process_port_expire(
                 event.timestamp, event.dp_name, event.port_no, event.eth_src, event.vid)),
-        ])
+        ]
+        if self._device_report_server:
+            handlers.extend([
+                (
+                    FaucetEvent.PortChange,
+                    lambda event: self._device_report_server.process_port_change(
```

Comment:
don't include this logic here -- keep it super simple. So, the handlers being added should be unconditional, and then any logic (conditional) should be handled in a sub method.  Yes, it adds a few (two?) helper methods, but the goal/intent is to keep the dispatch table very clean/simple.

Reply:
I updated. PTAL and see if my changes match what you had in mind.

Reply:
Not quite... What I was going for was basically that the dispatch table was static, and *always* had a port change event handler... and then the conditional logic was inside of the port change handler, i.e.:

`(FaucetEvent.PortChange, lambda event: self._process_port_chance(event))`

and then

```
def _process_port_change(self, event):
  if self._device_report_server:
    self.device_report_server.process_port_change(event)
```

or... create a "null" device_report_server that is instantiated if there's not a full one used. So, the method is always called, but sometimes it just does nothing.

Reply:
faucet events class was updated to accept multiple handlers for each event, so in this case FaucetEvent L2Learn and PortChange have handlers both from the device report server and the faucet state collector. Using your suggestion and have the multiple handlers support are going to be counter intuitive. what do you think?